pkg/cli/admin/catalog/build.go

```diff
@@ -0,0 +1,216 @@
+package catalog
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/operator-framework/operator-registry/pkg/apprclient"
+	"github.com/operator-framework/operator-registry/pkg/sqlite"
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/klog"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/util/templates"
+
+	imgappend "github.com/openshift/oc/pkg/cli/image/append"
+)
+
+func NewBuildImageOptions(streams genericclioptions.IOStreams) *BuildImageOptions {
+	return &BuildImageOptions{
+		IOStreams:           streams,
+		AppRegistryEndpoint: "https://quay.io/cnr",
```

Comment:
I'm not convinced this should be hardcoded/defaulted like so. Have a look at my other comment about defaulting `From`.

Reply:
In this case, quay.io is (as far as I know) the only publicly-accessible appregistry server, so I think it's a reasonable default.